src/components/organisms/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,113 @@
+import { CSSProperties, Dispatch, SetStateAction, useState } from 'react'
+import classes from './styles.module.scss'
+import classNames from 'classnames'
+import {
+  ExpandedState,
+  getExpandedRowModel,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+  PaginationState,
+  getPaginationRowModel,
+  TableMeta,
+  Row,
+  RowData,
+  ColumnDef,
+} from '@tanstack/react-table'
+import { Root } from '@radix-ui/react-form'
+import Container from 'components/atoms/Container/Container'
+import TablePagination from 'components/organisms/TablePagination/TablePagination'
+import TableHeaderGroup from 'components/organisms/TableHeaderGroup/TableHeaderGroup'
+
+export enum TableSizeTypes {
+  L = 'l',
+  M = 'm',
+  S = 's',
+}
+
+type DataTableProps<TData extends RowData> = {
+  data: TData[]
+  columns: ColumnDef<TData>[]
+  tableSize: TableSizeTypes
+  title?: string
+  onSortingChange?: (value: string | boolean, columnId: string) => void
+  onColumnFiltersChange?: (filters: string[], columnId: string) => void
+  pagination?: PaginationState
+  setPagination?: Dispatch<SetStateAction<PaginationState>>
```

Comment:
**[blocker]**

Would prefer if we didn't use the Dispatch<SetStateAction... types for functions. This limits the prop that we can pass here heavily. Basically this type tells us that this can only be used with react useState, which is an unnecessary limitation and in case of table there is a pretty big chance that we won't be storing the pagination data inside useState.

In similar cases prefer regular arrow function types. (state: PaginationState) => void for example